tendrl/gluster_integration/sds_sync/brick_device_details.py

```diff
@@ -0,0 +1,92 @@
+from tendrl.commons.event import Event
+from tendrl.commons.message import Message
+from tendrl.commons.utils import cmd_utils
+
+
+def get_brick_source_and_mount(brick_path):
+    # source and target correspond to fields "Filesystem" and
+    # "Mounted on" from df command output. The below command
+    # gives the filesystem and mount point for a given path,
+    # Eg: "/dev/mapper/tendrlMyBrick4_vg-tendrlMyBrick4_lv " \
+    #     "/tendrl_gluster_bricks/MyBrick4_mount"
+
+    command = "df --output=source,target " + brick_path.split(":")[-1]
+    cmd = cmd_utils.Command(command)
+    out, err, rc = cmd.run()
+    if rc != 0:
+        Event(
+            Message(
+                priority="error",
+                publisher=NS.publisher_id,
+                payload={
+                    "message": "%s command failed: %s" % (
+                        command, err
+                    )
+                }
+            )
+        )
+        return None, None
+    return out.split("\n")[-1].split()
+
+
+def get_disks(dev, disks):
+    # recursively find the physical devices under the
+    # given device
+    if dev.type == "disk":
+        if dev.path not in disks:
+            disks.append(dev.path)
+        return
+
+    for d in dev.ancestors:
+        get_disks(d, disks)
+
+    return
```

Comment:
In the case of a multipath, do you want the multipath (eg: `/dev/mapper/mpatha`) or the set of individual paths to the single underlying block device (eg: `/dev/sda`, `/dev/sdb`, &c)?